### PR TITLE
updated references of base plugins to netcommon 

### PR DESCRIPTION
**Pull request by KB-perByte**
##### SUMMARY

Minimum required ansible.netcommon version is 2.5.1
Updated base plugin references to ansible.netcommon

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/terminal/frr.py [hunk 1]
```diff
@@ -23,7 +23,9 @@
 import re
 
 from ansible.errors import AnsibleConnectionFailure
-from ansible.plugins.terminal import TerminalBase
+from ansible_collections.ansible.netcommon.plugins.plugin_utils.terminal_base import (
```
Comment (NilashishC) [inline plugins/terminal/frr.py:26]: Isn't this also needed for Cliconf?

Comment (KB-perByte) [reply plugins/terminal/frr.py:26]: @NilashishC  Added it, Thanks I missed it earlier.

**Review by NilashishC — APPROVED** [review]

File: changelogs/fragments/netcommon_ref_update.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+major_changes:
+  - "Minimum required ansible.netcommon version is 2.5.0."
```
Comment (Qalthos) [inline changelogs/fragments/netcommon_ref_update.yaml]:
```suggestion
  - "Minimum required ansible.netcommon version is 2.5.1."
```

File: galaxy.yml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 authors:
   - Ansible Network Community (ansible-network)
 dependencies:
-  "ansible.netcommon": "*"
+  "ansible.netcommon": ">=2.5.0"
```
Comment (Qalthos) [inline galaxy.yml]: This has to be 2.5.1 to work
```suggestion
  "ansible.netcommon": ">=2.5.1"
```

**Review by Qalthos — CHANGES_REQUESTED** [review]

**Review by Qalthos — APPROVED** [review]

**Pull request merged**